Lib/glyphsLib/classes.py

```diff
@@ -1207,6 +1207,12 @@ def __iter__(self):
         # instead of keys. It matches Glyphs.app though. Urg.
         yield from self._owner._userData.values()
 
+    def __copy__(self):
+        return dict(self)
+
+    def __deepcopy__(self, memo):
+        return {k: copy.copy(v) for k, v in self.items()}
```

Comment:
this isn't recursively "deep" copy, only one level deeper than shallow. Why do we need to customize `__copy__` and `__deepcopy__` for this class?

Reply:
So I first used deepcopy, then I found that `Proxy` (the parent class) has implmentations of `__copy__` and `__deepcopy__` that return lists (and `__deepcopy__` even class a non-existent `.copy() of the items), so I went ahead and tried to override in in `UserDataProxy`, missing the forest for the trees...